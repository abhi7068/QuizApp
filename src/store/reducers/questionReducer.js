import * as actionTypes from "../actions/actionTypes";
const initialState = {
    quiz: [],
};
const questionReducer = (state = initialState, action) => {
    console.log(action);
    switch (action.type) {
        case actionTypes.ADD_QUESTION:
            console.log(action)
            return {
                ...state,
              quiz:[...state.quiz,action.question]
            };
        default:
            return state;
    }
};

export default questionReducer;