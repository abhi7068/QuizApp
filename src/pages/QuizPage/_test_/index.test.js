import React from 'react'
import QuizPage from '../../QuizPage/index'
import {render,cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import questionReducer from '../../../store/reducers/questionReducer'
import {BrowserRouter as Router,withRouter} from 'react-router-dom';
const store= createStore(questionReducer);
afterEach(cleanup);
it('matches snapshot',()=>{
const {asFragment} = render(
<Provider store={store}>
          <Router>
           <QuizPage />
        </Router>
      </Provider>
    );
expect(asFragment()).toMatchSnapshot();
});
 
it('it renders',()=>{
render(
<Provider store = {store}>
          <Router>
       <QuizPage/>
        </Router>
      </Provider>
  );
});

