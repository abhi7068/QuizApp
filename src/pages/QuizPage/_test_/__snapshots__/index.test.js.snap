// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <h2>
      No questions available
    </h2>
  </div>
</DocumentFragment>
`;
